resources/aws/auto_scaling_group.go

```diff
@@ -161,3 +163,85 @@ func (asg *AutoScalingGroup) findExisting() (*autoscaling.Group, error) {
 
 	return autoScalingGroups.AutoScalingGroups[0], nil
 }
+
+// FindLegacy returns true if there is an ASG created for the same cluster name
+// but not being part of a CloudFormation stack.
+func (asg *AutoScalingGroup) FindLegacy() (bool, error) {
+	a, err := asg.findExisting()
+	if IsNotFound(err) {
+		return false, nil
+	} else if err != nil {
+		return false, microerror.Mask(err)
+	}
+
+	var keyNameFound, keyClusterFound bool
+	for _, td := range a.Tags {
+		if *td.Key == tagKeyName && *td.Value == asg.Name {
```

Comment:
We could improve this a bit by separating network related actions and standalone algorithms. I am thinking about having a method that does the name, key and prefix checking. This method can then be unit tested and the code here is more easily understandable. Then we also only have one boolean we deal with here.

Reply:
Good point! Done, pls take a look